glpi-project/glpi
# Routing, templates, bootstrap, ...

| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | yes
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #4594 #3392 #4593 #2925 

Currently old display system is still not affected with most of the changes (excluding DI).

What has been done:
- symfony DI for dependency injection (see related ticket),
- Slim framework, to handle routing,
- twig as templating system,
- use search results to display sub-lists almost everywhere,
- add a form build system based on templates and their inheritance, so building forms can be automated in most cases,
- use bootstrap on UI based on adminlte,
- use a JS dependency manager (bower for instance, the first I've been able to install and get working, not really a choice)
- use tracy as debug bar (to give it a try, it would be possible to add current debug tabs as tracy's panels)

There is still a lot of work to get the new system entirely working, such as a Middleware to handle routes authentication, add display hooks for plugins, manage all specific cases/pages/whatever (there are lots of them in itil objects), etc.

I'm not planning to work on the port of the API, I'm not sure it should be (maybe would it be simplier to rework it with slim possibilities).
